test/common/http/http2/codec_impl_test.cc

```diff
@@ -598,6 +602,13 @@ TEST_P(Http2CodecImplTest, SimpleRequestResponse) {
   }
 }
 
+TEST_P(Http2CodecImplTest, ClientUnexpectedHeaders) {
+  initialize();
+
+  Http::Status status = Http2CodecImplTestFixture::onConnBeginHeaders(client_.get(), 3);
+  EXPECT_FALSE(status.ok());
```

Comment:
can you instead verify status.message includes parts of your error string to verify it's the expected error?
thanks!
/wait

Reply:
https://github.com/envoyproxy/envoy/pull/36573/commits/f2e38e16c9205d113aeab91c9977c22508abf2b3